autoarena/service/head_to_head.py

```diff
@@ -84,38 +93,45 @@ def get_count(project_slug: str) -> int:
 
```

Comment:
Noticed that `HeadToHeadService.get_count` is a fair bit slower on larger datasets (~100k head-to-heads). Taking a look at perf here next.